### PR TITLE
feat: take into account TaskDefs with only TestSelectors.

### PR DESCRIPTION
**Pull request by kpodsiad**
`TestSelector` can be used to create `TaskDef` which should only execute selected tests from the given test class.

More info can be found at https://github.com/sbt/junit-interface/pull/108 (the changes are almost the same)

~Could you point me how to add a test for that feature and for compatibility?~
That was a silly question, I've already done tests for this in Junit, why not use them here as well.

tl;dr why
This will allow Metals to run a single munit test from the whole suite.

### CODE REVIEW

File: tests/jvm/src/test/scala/munit/TestSelectorSuite.scala
Comment (kpodsiad) [inline tests/jvm/src/test/scala/munit/TestSelectorSuite.scala:39]: So this test mimics the behavior of sbt's `ForkMain` - it creates a framework instance which later will be used for running tests.

File: tests/jvm/src/test/scala/munit/TestSelectorSuite.scala
Comment (kpodsiad) [inline tests/jvm/src/test/scala/munit/TestSelectorSuite.scala:54]: Each executed test will send an event, so I use that fact to gather information about executed tests.

File: junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java
Comment (olafurpg) [inline junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java:173]: Should we guard this cast?

**Review by olafurpg — APPROVED** [review]
LGTM! It’s very nice to have this functionality

File: junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java
Comment (kpodsiad) [reply junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java:173]: I think I can try to get rid off this cast

File: junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java
Comment (kpodsiad) [inline junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java:179]: @olafurpg this is the best what can I do in Java 😅 At least now, casting is done just after `instanceof` check

Comment (olafurpg) [reply junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java:179]: LGTM 👍

Comment (olafurpg) [reply junit-interface/src/main/java/munit/internal/junitinterface/JUnitRunner.java:179]: I think you can replace the if/else with a ternary operator to drop the block expression with return statements

**Comment by kpodsiad** [comment]
I've applied suggestions and squashed commits - the PR should be mergeable now.
Could you publish the new version with these changes after merging @olafurpg?

**Comment by olafurpg** [comment]
@kpodsiad As a member of the Scalameta org you have merge rights. Feel free to merge and push a git tag v1.0.0-M3 to trigger a release. I use the following scripts locally so I can execute `git release VERSION`
```
❯ cat `which git-release.sh`
set -eux
TAG="v$1"
echo $TAG
git tag -fa "$TAG" -m $TAG && git push -f origin $TAG

lsif-typescript on 🌱 cross-project via 🐳 default is 📦 v0.1.14 via ⬢ v16.7.0 on ☁️ olafurpg@sourcegraph.com
❯ cat ~/.gitconfig | grep git-release
	release = !git-release.sh $2
❯ git release 1.0.0-M3
```

**Comment by kpodsiad** [comment]
I didn't want to abuse my membership in the project not related directly to Metals ;)

**Pull request merged**
